build-and-run 运行报错

zyf@zyfdeiMac ~ % /Users/zyf/Desktop/Project/simple-master/build-and-run ; exit;
curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused
bash: -c: line 0: syntax error near unexpected token `;'
bash: -c: line 0: `; bashmatic-install'
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 24: /Users/zyf/.bashmatic/init.sh: No such file or directory
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 32: h1.purple: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 36: h1: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 37: h1: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 38: h1: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 42: hl.subtle: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 43: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 52: hl.subtle: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 53: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 54: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 55: run.set-next: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 56: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 57: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 58: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 62: hl.subtle: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 63: run.set-all: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 64: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 65: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 66: run: command not found
/Users/zyf/Desktop/Project/simple-master/build-and-run: line 71: error: command not found

[进程已完成]

手动编译也报错



zyf@zyfdeiMac ~ % cd /Users/zyf/Desktop/Project/simple-master 
zyf@zyfdeiMac simple-master % mkdir build
zyf@zyfdeiMac simple-master % cd build
zyf@zyfdeiMac build % cmake ..
-- The C compiler identification is AppleClang 12.0.0.12000032
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/zyf/Desktop/Project/simple-master/build/test/googletest-download
Scanning dependencies of target googletest
[ 11%] Creating directories for 'googletest'
[ 22%] Performing download step (download, verify and extract) for 'googletest'
-- Downloading...
   dst='/Users/zyf/Desktop/Project/simple-master/build/test/googletest-download/googletest-prefix/src/release-1.10.0.zip'
   timeout='none'
-- Using src='https://github.com/google/googletest/archive/release-1.10.0.zip'
-- [download 100% complete]
-- Downloading... done
-- extracting...
     src='/Users/zyf/Desktop/Project/simple-master/build/test/googletest-download/googletest-prefix/src/release-1.10.0.zip'
     dst='/Users/zyf/Desktop/Project/simple-master/build/test/googletest-src'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 33%] No patch step for 'googletest'
[ 44%] No update step for 'googletest'
[ 55%] No configure step for 'googletest'
[ 66%] No build step for 'googletest'
[ 77%] No install step for 'googletest'
[ 88%] No test step for 'googletest'
[100%] Completed 'googletest'
[100%] Built target googletest
-- Found PythonInterp: /usr/bin/python (found version "2.7.16") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE  
CMake Warning (dev) at /Users/zyf/Library/Android/sdk/cmake/3.10.2.4988404/share/cmake-3.10/Modules/GoogleTest.cmake:254 (if):
  Policy CMP0057 is not set: Support new IN_LIST if() operator.  Run "cmake
  --help-policy CMP0057" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  IN_LIST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  test/CMakeLists.txt:44 (gtest_add_tests)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /Users/zyf/Library/Android/sdk/cmake/3.10.2.4988404/share/cmake-3.10/Modules/GoogleTest.cmake:254 (if):
  if given arguments:

    "TARGET" "IN_LIST" "allKeywords"

  Unknown arguments specified
Call Stack (most recent call first):
  test/CMakeLists.txt:44 (gtest_add_tests)


-- Configuring incomplete, errors occurred!
See also "/Users/zyf/Desktop/Project/simple-master/build/CMakeFiles/CMakeOutput.log".

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.2)
 project(simple_tests)
 
+# https://github.com/maps-gpu/MAPS/issues/7#issuecomment-418200278
+cmake_policy(SET CMP0057 NEW)
 # Download and unpack googletest at configure time
 configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
 execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .

--- a/test/CMakeLists.txt.in
+++ b/test/CMakeLists.txt.in
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.2)
+cmake_policy(SET CMP0057 NEW)
 
 project(googletest-download NONE)
 

